test/Confluent.Kafka.IntegrationTests/Tests/Producer_Produce.cs

```diff
@@ -28,6 +32,63 @@ namespace Confluent.Kafka.IntegrationTests
     /// </summary>
     public partial class Tests
     {
+        [Theory]
+        [MemberData(nameof(KafkaParameters))]
+        public void Producer_Produce_WithOpenTelemetryInstrumentation(string bootstrapServers)
+        {
+            LogToFile("start Producer_Produce_WithOpenTelemetryInstrumentation");
+
+            // Prepare the activity events listener
+            string activityName = "Confluent.Kafka.MessageProduced";
+            ActivityEventsRecorder eventsRecorder = new(activityName);
+            ActivityListener listener = eventsRecorder.BuildActivityListener();

```

Comment:
Another alternative way to test would be to use OpenTelemetry's InMemoryExporter. Example : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlClientTests.cs#L90